crates/bevy_ecs/src/entity/clone_entities.rs

```diff
@@ -0,0 +1,453 @@
+            ignored_components,
+            ignored_bundles,
+            allowed_components,
+            allowed_bundles,
+            clone_handlers_overrides,
+            ..
+        } = self;
+
+        let mut component_clone_handlers = ComponentCloneHandlers::default();
+        for (k, v) in clone_handlers_overrides.into_iter() {
+            if let Some(component_id) = world.components().get_id(k) {
+                component_clone_handlers.set_component_handler(component_id, v);
+            };
+        }
+
+        let mut ignored_components = ignored_components
+            .into_iter()
+            .flat_map(|type_id| world.components().get_id(type_id))
+            .collect::<HashSet<_>>();
+        for getter in ignored_bundles {
+            (getter)(world.components(), &mut ignored_components);
+        }
+
+        let allowed = !allowed_components.is_empty();
+        let filter = if allowed {
+            let mut allowed_components = allowed_components
+                .into_iter()
+                .flat_map(|type_id| world.components().get_id(type_id))
+                .filter(|component_id| !ignored_components.contains(component_id))
+                .collect::<HashSet<_>>();
+            for getter in allowed_bundles {
+                (getter)(
+                    world.components(),
+                    &mut allowed_components,
+                    &ignored_components,
+                );
+            }
+            allowed_components
+        } else {
+            ignored_components
+        };
+
+        EntityCloner {
+            source,
+            target,
+            allowed,
+            filter: &filter,
+            clone_handlers_overrides: &component_clone_handlers,
+        }
+        .clone_entity(world);
+    }
+
+    /// Add a component to the list of components to clone.
+    /// Calling this function automatically disallows all other components, only explicitly allowed ones will be cloned.
+    pub fn allow<T: Component>(&mut self) -> &mut Self {
+        self.allowed_components.insert(TypeId::of::<T>());
+        self
+    }
+
+    /// Extend the list of components to clone.
+    /// Calling this function automatically disallows all other components, only explicitly allowed ones will be cloned.
+    pub fn allow_by_ids(&mut self, ids: impl IntoIterator<Item = TypeId>) -> &mut Self {
+        self.allowed_components.extend(ids);
+        self
+    }
+
+    /// Reset the filter to allow all components to be cloned
+    pub fn allow_all(&mut self) -> &mut Self {
+        self.allowed_components.clear();
+        self.allowed_bundles.clear();
+        self.ignored_components.clear();
+        self.ignored_bundles.clear();
+        self
+    }
+
+    /// Add a bundle of components to the list of components to clone.
+    /// Calling this function automatically disallows all other components, only explicitly allowed ones will be cloned.
+    pub fn allow_bundle<T: Bundle>(&mut self) {
```

Comment:
Agreed, I just though that storing a closure instead of an id just to allow a single component might not be optimal, but if builder is changed to store `ComponentId`s anyway using `World` that should be fine.